use_sim_time: true

# basic program type
# refer to src/legged_ctrl/include/LeggedState.h
run_type: 0
robot_type: 1     # 0: A1
                  # 1: Go1
mpc_type: 1       # 0: CI-MPC
                  # 1: convex MPC 
                  # 2: ocs2 nonlinear MPC
low_level_type: 0 # 0: basic tau ctrl
                  # 1: whole body control 
# default foothold position in robot body frame (varialbes with "_rel" in the code)
default_foot_pos_FL_x: 0.17
default_foot_pos_FL_y: 0.12
default_foot_pos_FL_z: -0.3

default_foot_pos_FR_x: 0.17
default_foot_pos_FR_y: -0.17
default_foot_pos_FR_z: -0.3

default_foot_pos_RL_x: -0.17
default_foot_pos_RL_y: 0.17
default_foot_pos_RL_z: -0.3

default_foot_pos_RR_x: -0.17
default_foot_pos_RR_y: -0.12
default_foot_pos_RR_z: -0.3

# gait speed in phase per second
gait_counter_speed: 4.0    # 3 gait cycle per second

# convex MPC parameters

q_weights_0:  50.0 # roll
q_weights_1:  100.0 # pitch
q_weights_2:  0.0 # yaw

q_weights_3:  0.0 # pos x
q_weights_4:  0.0 # pos y
q_weights_5:  3500.0 # pos z

q_weights_6:  0.01 # omega x
q_weights_7:  0.01 # omega y
q_weights_8:  10.0 # omega z

q_weights_9:  15.0 # vel x
q_weights_10: 15.0 # vel y
q_weights_11: 20.0 # vel z

q_weights_12: 0.0

r_weights_0:  0.0001
r_weights_1:  0.0001
r_weights_2:  0.0001

r_weights_3:  0.0001
r_weights_4:  0.0001
r_weights_5:  0.0001

r_weights_6:  0.0001
r_weights_7:  0.0001
r_weights_8:  0.0001

r_weights_9:  0.0001
r_weights_10: 0.0001
r_weights_11: 0.0001

# swing leg parameters
kp_foot_x: 0.5
kp_foot_y: 0.5
kp_foot_z: 0.5

kd_foot_x: 0.3
kd_foot_y: 0.3
kd_foot_z: 0.3

km_foot_x: 0.1
km_foot_y: 0.1
km_foot_z: 0.1




a1_robot_mass: 13.0

a1_trunk_inertia_xx: 0.0158533
a1_trunk_inertia_xy: 0.0
a1_trunk_inertia_xz: 0.0
a1_trunk_inertia_yz: 0.0
a1_trunk_inertia_yy: 0.0377999
a1_trunk_inertia_zz: 0.0456542